src/structures/core.ts

```diff
@@ -0,0 +1,154 @@
+import type { Options } from "lru-cache"
+import type { RequestInit } from "node-fetch"
+
+export type Version = 1 | 2
+
+export type ValueOf<T> = T[keyof T]
+export type Nullable<T> = T | null
+
+export interface BotStates {
+    NONE: 0
+    OFFICIAL: 1
+    KOREANBOTS_VERIFIED: 4
+    PARTNER: 8
+    DISCORD_VERIFIED: 16
+    PREMIUM: 32
+    HACKATHON_BOT: 64
+}
+
+export interface UserFlags {
+    NONE: 0,
+    ADMINISTRATOR: 1
+    BUG_HUNTER: 2
+    BOT_REVIEWER: 4
+    PREMIUM_USER: 8
+}
```

Comment:
How about using enum

Reply:
I’ll consider about this later because at the bottom of file it’s using ValueOf<UserFlags> to type flags of bots/users